src/handler.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+var uniq = require('lodash.uniq');
+
+function Handler(retriever) {
+  this._fetched = false;
+  this._page = 1;
+  this._initRequest = false;
+  this._retriever = retriever;
+  this._totalPages = 1;
+  this.hits = [];
+  this.lastPage = false;
+}
+
+Handler.prototype._handleResponse = function(response) {
+  this.response = response;
+  var data = response.body;
+
+  this.totalHits = data.totalHits;
+  this.hits = uniq(this.hits.concat(data.hits));
+
+  var per_page = this._retriever._query.per_page || 20;
+  this._totalPages = Math.ceil(this.totalHits / per_page);
+
+  if (this._page === this._totalPages) {
+    this.lastPage = true;
+  }
+
+  return response;
+};
+
+Handler.prototype.next = function() {
+  if (!this._initRequest) {
+    throw new Error('No initial request has been made');
+  }
+
+  if (this.lastPage) {
+    throw new Error('Last page has been retrieved');
```

Comment:
Return a promise for `null`, this will be consistent with other APIs we've written. Also, in cases where you do need to throw an error, and you in a promise returning function, it is more appropriate to return a promise in the failure state, than synchronously throwing an error.